firefox_puppeteer/api/software_update.py

```diff
@@ -0,0 +1,544 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import ConfigParser
+import os
+import re
+
+import mozinfo
+
+from ..base import BaseLib
+from .appinfo import AppInfo
+from .prefs import Preferences
+
+
+class SoftwareUpdate(BaseLib):
+    """The SoftwareUpdate API adds support for an easy access to the update process."""
+
+    PREF_APP_DISTRIBUTION = 'distribution.id'
+    PREF_APP_DISTRIBUTION_VERSION = 'distribution.version'
+    PREF_APP_UPDATE_URL = 'app.update.url'
+    PREF_DISABLED_ADDONS = 'extensions.disabledAddons'
+
+    def __init__(self, marionette_getter):
+        BaseLib.__init__(self, marionette_getter)
+
+        self.app_info = AppInfo(marionette_getter)
+        self.prefs = Preferences(marionette_getter)
+
+        self.update_channel = UpdateChannel(self)
+        self.mar_channels = MARChannels(self)
+        self.active_update = ActiveUpdate(self)
+
+        self._aus = self.marionette.execute_script("""
+            return Cc['@mozilla.org/updates/update-service;1']
+                .getService(Ci.nsIApplicationUpdateService);
+            """)
+
+    @property
+    def ABI(self):
+        """Get the customized ABI for the update service.
+
+        :returns: ABI version
+        """
+
+        abi = self.app_info.XPCOMABI
+        if mozinfo.isMac:
+            abi += self.marionette.execute_script("""
+                let macutils = Cc['@mozilla.org/xpcom/mac-utils;1']
+                    .getService(Ci.nsIMacUtils);
+                if (macutils.isUniversalBinary)
+                    return '-u-' + macutils.architecturesInBinary;
+
+                return '';
+            """)
+
+        return abi
+
+    @property
+    def allowed(self):
+        """Check if the user has permissions to run the software update
+
+        :returns: Status if the user has the permissions
+        """
+
+        return self._aus['canCheckForUpdates'] and self._aus['canApplyUpdates']
+
+    def assert_update_applied(self, update_data):
+        """Checks if an update has been applied correctly.
+
+        :param update_data: All the data collected during the update process
+        """
+
+        # Get the information from the last update
+        info = update_data['updates'][update_data['update']['index']]
+
+        # The upgraded version should be identical with the version given by
+        # the update and we shouldn't have run a downgrade
+        check = self.marionette.execute_script("""
+            Cu.import("resource://gre/modules/Services.jsm");
+
+            return  Services.vc.compare(arguments[0], arguments[1]);
+        """, script_args=[info['build_post']['version'], info['build_pre']['version']])
+
+        assert check >= 0, 'The version of the upgraded build is higher or equal'
```

Comment:
That is not what is currently being done in `prefs` (e.g., [1])  or `windows` [2]. I can make that change if you wish, but I wanted to check with you first for consistency.

[1] https://github.com/mozilla/firefox-ui-tests/blob/master/firefox_puppeteer/api/prefs.py#L153
[2] https://github.com/mozilla/firefox-ui-tests/blob/master/firefox_puppeteer/ui/windows.py#L318